frappe/database/database.py

```diff
@@ -769,7 +769,16 @@ def table_exists(self, doctype):
 		return ("tab" + doctype) in self.get_tables()
 
 	def get_tables(self):
-		return [d[0] for d in self.sql("select table_name from information_schema.tables where table_schema not in ('pg_catalog', 'information_schema')")]
+		tables = frappe.cache().get_value('db_tables')
+		if not tables:
+			table_rows = self.sql("""
+				SELECT table_name
+				FROM information_schema.tables
+				WHERE table_schema NOT IN ('pg_catalog', 'information_schema')
+			""")
+			tables = set(d[0] for d in table_rows)
+		frappe.cache().set_value('db_tables', tables)
```

Comment:
shouldn't this be tabbed in?

since you'll only set it when you don't have the cache set, otherwise it is pretty redundant